test/integration/controlplane/transformation/kmsv2_transformation_test.go

```diff
@@ -208,6 +212,84 @@ resources:
 	}
 }
 
+// TestKMSv2ProviderKeyIDStaleness is an integration test between KubeAPI and KMSv2 Plugin
+// Concretely, this test verifies the following integration contracts:
+// 1. A no-op update on a single item and when the key IDs are the same the resource version should not change
+// 2. When key ID changes the resource version changes (but only once)
+func TestKMSv2ProviderKeyIDStaleness(t *testing.T) {
```

Comment:
At the end of this test, I think we should disable the plugin, confirm that creating a _new_ secret fails, but that a no-op update does not fail.